### ISSUE TITLE
VinylCallbackOnStop has issues

### ISSUE THREAD

**Issue by FoolsLynx** [JujuAdams/Vinyl#111]
I was playing around with VinylCallbackOnStop and discovered that there seems to be a few issues. 

1. In __VinylSystem on line 251, __IsPlaying() is not set. This is fixed by changing:
`if (not _callbackArray[_i].__IsPlaying())`
to
`if (not _callbackArray[_i].__voiceStruct.__IsPlaying())`

2. When trying to detach a callback from a voice, the callback never seems to be removed and ends up in a complete loop. (haven't found a fix)

Rough Code:

```
// Create
global.music_track = VinylPlay(musMain, true, 1.0, 1.0);

// On Space Pressed
VinylCallbackOnStop(global.music_track, function() {
    VinylCallbackOnStop(global.music_track, undefined);
    VinylStop(global.music_track);
});
```

Regardless of Space being pressed only once, the function continues endlessly and for whatever reason despite both global.music_track values being the same, the one detaching the callback is seen by vinyl as completely different and returns before trying to call the callback.

**Comment by JujuAdams** [comment]
What version of Vinyl are you using?

**Comment by FoolsLynx** [comment]
6.0.8 Beta

I mainly combined them as I'm not sure if the second issue is linked to the first or not as without the fix I did in the first place, I wouldn't have been able to do anything with the second regardless.

**Comment by JujuAdams** [comment]
Fixed for 6.0.9
